Content.Server/_CD/Admin/EventPreferencePanelEui.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Content.Server.Administration;
+using Content.Server.EUI;
+using Content.Shared._CD.Admin;
+using Content.Shared.Eui;
+using Content.Shared.Preferences;
+using Content.Shared.Roles;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server._CD.Admin;
+
+public sealed class EventPreferencesPanelEui : BaseEui
+{
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+
+    private readonly HumanoidCharacterProfile _playerPref;
+    private readonly LocatedPlayerData _targetPlayer;
+    private List<AntagPrototype> _visibleAntagPrototypes = new List<AntagPrototype>();
+
+    public EventPreferencesPanelEui(LocatedPlayerData player, HumanoidCharacterProfile pref)
+    {
+        IoCManager.InjectDependencies(this);
+        _targetPlayer = player;
+        _playerPref = pref;
+    }
+
+    public override EuiStateBase GetNewState()
+    {
+        return new EventPreferencePanelEuiState(
+            _targetPlayer.Username,
+            _playerPref,
+            _visibleAntagPrototypes
+            );
+    }
+
+    public async void SetState()
+    {
+        var visibleAntagPrototypes = new List<AntagPrototype>();
+        foreach (var antagPrototype in _proto.EnumeratePrototypes<AntagPrototype>())
+        {
+            if(antagPrototype.VisiblePreference)
+                visibleAntagPrototypes.Add(antagPrototype);
+        }
+
+        _visibleAntagPrototypes = visibleAntagPrototypes;
+
+        StateDirty();
+    }
```

Comment:
It should not be needed to manually call this method when the UI is opened. Given that this *should* be constant within a round it can probably be set within the constructor?